test/topic_statistics_collector/test_received_message_period.cpp

```diff
@@ -38,6 +42,50 @@ constexpr const double kExpectedMaxMilliseconds{1000.0};
 constexpr const double kExpectedStandardDeviation{0.0};
 }  // namespace
 
+TEST(ReceivedMessagePeriodTest, TestUntypedPeriodMeasurement) {
+  ReceivedUntypedMessagePeriodCollector test{};
+  rmw_message_info_t msg_info = rmw_get_zero_initialized_message_info();
+
+  EXPECT_FALSE(test.IsStarted()) << "Expected to be not started after constructed";
+
+  EXPECT_TRUE(test.Start()) << "Expected Start() to be successful";
+  EXPECT_TRUE(test.IsStarted()) << "Expected to be started";
+
+  rcl_time_point_value_t fake_now_nanos_{1};
+  msg_info.received_timestamp = fake_now_nanos_;
+
+  test.OnMessageReceived(msg_info);
+  auto stats = test.GetStatisticsResults();
+  EXPECT_EQ(0, stats.sample_count) << "Expected 0 samples to be collected";
+
+  fake_now_nanos_ +=
+    std::chrono::duration_cast<std::chrono::nanoseconds>(kDefaultDurationSeconds).count();
+  msg_info.received_timestamp = fake_now_nanos_;
+
+  test.OnMessageReceived(msg_info);
+  stats = test.GetStatisticsResults();
+  EXPECT_EQ(1, stats.sample_count) << "Expected 1 sample to be collected";
+
+  fake_now_nanos_ +=
+    std::chrono::duration_cast<std::chrono::nanoseconds>(kDefaultDurationSeconds).count();
+  msg_info.received_timestamp = fake_now_nanos_;
+
+  test.OnMessageReceived(msg_info);
+  stats = test.GetStatisticsResults();
+  EXPECT_EQ(2, stats.sample_count) << "Expected 2 samples to be collected";
+
+  fake_now_nanos_ +=
+    std::chrono::duration_cast<std::chrono::nanoseconds>(kDefaultDurationSeconds).count();
+  msg_info.received_timestamp = fake_now_nanos_;
+
+  test.OnMessageReceived(msg_info);
+  stats = test.GetStatisticsResults();
+  EXPECT_EQ(3, stats.sample_count);
```

Comment:
The same comment 
This block of code could be simplified with loop and index variable the same way as it is made in the TestMessagesWithDifferentDefaultTimes test.
@CursedRock17 Could you please refactor it?